src/ray/gcs/gcs_actor_manager.cc

```diff
@@ -276,7 +276,7 @@ void GcsActorManager::HandleReportActorOutOfScope(
 
     DestroyActor(actor_id,
                  GenActorOutOfScopeCause(actor),
-                 /*force_kill=*/true,
+                 /*force_kill=*/false,
```

Comment:
@jjyao do you recall why`force_kill` was set to true when actors go out of scope? It was done in https://github.com/ray-project/ray/commit/077376021e939b36bdc49fcc19b486928f001411

One reason maybe to avoid complexity and cleanup fast because graceful shutdown could hang. We can consider adding a timeout to deal with that. But, is there some other reason that i'm missing?

Reply:
there will be a potentially arbitrary delay for the `__ray_shutdown__` method now though... I think we'll need to introduce some kind of timeout for this that has a reasonable default (`30s`?) and can be overridden by users

we'll also need to implement fallback to force kill the actor here.